build/yaml/crd/vpc/crd.nsx.vmware.com_subnets.yaml

```diff
@@ -163,10 +180,25 @@ spec:
                 && has(self.advancedConfig.staticIPAllocation.enabled) && self.advancedConfig.staticIPAllocation.enabled==true
                 && has(self.subnetDHCPConfig) && has(self.subnetDHCPConfig.mode) &&
                 (self.subnetDHCPConfig.mode==''DHCPServer'' || self.subnetDHCPConfig.mode==''DHCPRelay''))'
+            - message: DHCPServerAddresses can only be set when DHCP mode is DHCPServer
+              rule: '!(has(self.advancedConfig) && has(self.advancedConfig.dhcpServerAddresses)
+                && size(self.advancedConfig.dhcpServerAddresses)>0 && (!has(self.subnetDHCPConfig)
+                || !has(self.subnetDHCPConfig.mode) || self.subnetDHCPConfig.mode!=''DHCPServer''))'
+            - message: GatewayAddresses can only be set when DHCP mode is DHCPServer
```

Comment:
GatewayAddresses can be specified for non DHCP case, right? @lxiaopei

Reply:
fixed